File: src/models/Pendente.java
package models;

public class Pendente {
    private String nome;
    private String numeroTelefone;

    public Pendente() {
        this.nome = "";
        this.numeroTelefone = "";
    }

    public Pendente(String nome, String numeroTelefone) {
        this.nome = nome;
        this.numeroTelefone = numeroTelefone;
    }

    public Pendente(Pendente outro) {
        this.nome = outro.nome;
        this.numeroTelefone = outro.numeroTelefone;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNumeroTelefone() {
        return numeroTelefone;
    }

    public void setNumeroTelefone(String numeroTelefone) {
        this.numeroTelefone = numeroTelefone;
    }

    @Override
    public Pendente clone() {
        return new Pendente(this);
    }

    @Override
    public String toString() {
        return "Pendente{" +
                "nome='" + nome + '\'' +
                ", numeroTelefone='" + numeroTelefone + '\'' +
                '}';
    }

    
}

File: Python/backend/models/Pendente.py
class Pendente:
    def __init__(self, nome="", numero_telefone="", outro=None):
        """
        Construtor da classe Pendente
        Pode ser usado como:
        - Pendente() - construtor padrão
        - Pendente(nome, numero_telefone) - construtor com parâmetros
        - Pendente(outro=outro_pendente) - construtor de cópia
        """
        if outro is not None:
            # Construtor de cópia
            self.nome = outro.nome
            self.numero_telefone = outro.numero_telefone
        else:
            # Construtor padrão ou com parâmetros
            self.nome = nome
            self.numero_telefone = numero_telefone

    # Getters e Setters
    def get_nome(self):
        return self.nome

    def set_nome(self, nome):
        self.nome = nome

    def get_numero_telefone(self):
        return self.numero_telefone

    def set_numero_telefone(self, numero_telefone):
        self.numero_telefone = numero_telefone

    def clone(self):
        return Pendente(outro=self)
